### PR TITLE
Ensure disks removal after removing cluster in GKE

### PR DESCRIPTION
**Pull request by artemnikitin**
Thanks to @barkbay, there was found that after killing clusters in GKE disks from those instances are not deleted automatically. This change introducing a step for all CI jobs running in GKE to remove disks properly.

close https://github.com/elastic/cloud-on-k8s/issues/1162

### CODE REVIEW

File: build/ci/delete_unused_disks.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    content = f.read()
+    try:
+        parsed_json_dict = json.loads(content)
+        if len(parsed_json_dict) == 0:
+            print("There is no unused disks. Congratulations!")
+        else:
+            for entry in parsed_json_dict:
+                name = entry['name']
+                head, tail = os.path.split(entry['zone'])
+                os.system('gcloud compute disks delete {} --project {} --zone {} --quiet'
```
Comment (sebgl) [inline build/ci/delete_unused_disks.py]: I'm a bit more familiar with using the `subprocess` Python module. Does `os.System` display stdout if the command fails?

File: build/ci/delete_unused_disks.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env python
+
+import os
+import json
+
+project = os.environ['GCLOUD_PROJECT']
+
+os.system('gcloud compute disks list --filter="-users:*" --format="json" --project {} > unused_disks.json'
```
Comment (sebgl) [inline build/ci/delete_unused_disks.py]: Does this filter out disks that are currently being used?

Comment (artemnikitin) [reply build/ci/delete_unused_disks.py]: Yes, it's filtering out all disks which have object `users`. This object contains reference to an instance which uses disk. If disk is not used, then there will be no `users`

File: build/ci/delete_unused_disks.py (hunk 1 above)
Comment (artemnikitin) [reply build/ci/delete_unused_disks.py]: Output will be similar to what you can see in command line. At least I tried different commands and saw the same output as in command line.

**Comment by artemnikitin** [comment]
Looks like `infra` is doing something similar to this. I will contact them to sync our efforts, maybe they can do that on their side.

**Comment by artemnikitin** [comment]
`infra` solution is only for them.

File: build/ci/Makefile [hunk 3]
```diff
@@ -144,3 +144,15 @@ ci-e2e-delete-cluster: vault-gke-creds
     	-e "GKE_SERVICE_ACCOUNT_KEY_FILE=$(GO_MOUNT_PATH)/build/ci/$(GKE_CREDS_FILE)" \
     	cloud-on-k8s-ci-e2e \
     	bash -c "make -C operators set-context-gke delete-gke"
```
Comment (sebgl) [inline build/ci/Makefile]: IIUC, everytime we delete a gke cluster, the underlying disks are not deleted. This applies to e2e tests but also to our own clusters, right?
Should we do this everytime, as part of the `delete-gke` target?

Comment (artemnikitin) [reply build/ci/Makefile]: This fix applies to all disks, not only from clusters for e2e tests. No need to explicitly run `delete-gke`

Comment (sebgl) [reply build/ci/Makefile]: OK. But isn't there a bug in the `delete-gke` target (that we use ourselves) if the disks are not deleted?

Comment (artemnikitin) [reply build/ci/Makefile]: Yeah sure, if it runs and some unused disks are left, then it's a bug. I only run it before submitting original PR and it cleaned up everything at that moment.

**Comment by artemnikitin** [comment]
jenkins test this please

**Comment by thbkrkr** [comment]
I would like to understand what is the cause of this issue.
I just deleted a GKE cluster and watched the disks at the same time and the disks were automatically deleted. The statuses of the disks are immediately passed into `DELETING`.

The disks statuses were 
```bash
> gcloud compute disks list  | grep thb
gke-thb-e2e-cluster-default-pool-5b7e16fd-ppr0                   europe-west1-d  zone            30       pd-ssd       DELETING
gke-thb-e2e-cluster-default-pool-f3c9866f-rw75                   europe-west1-b  zone            30       pd-ssd       DELETING
gke-thb-e2e-cluster-default-pool-106593c7-3r5n                   europe-west1-c  zone            30       pd-ssd       DELETING
```

**Comment by thbkrkr** [comment]
> I would like to understand what is the cause of this issue.

I think the reason is that PVs were not deleted before the cluster was deleted.

**Comment by thbkrkr** [comment]
jenkins test this please

**Comment by artemnikitin** [comment]
jenkins test this please

**Pull request merged**